File: backend/api/src/main/java/kpn/api/common/RouteScope.java
package kpn.api.common;

import com.google.common.collect.ImmutableList;

public enum RouteScope {
  LOCAL,
  REGIONAL,
  NATIONAL,
  INTERNATIONAL,
  UNKNOWN;

  private static final ImmutableList<RouteScope> all = ImmutableList.of(
    LOCAL,
    REGIONAL,
    NATIONAL,
    INTERNATIONAL
  );

  public static final ImmutableList<RouteScope> all() {
    return all;
  }
}

File: backend/api/src/main/scala/kpn/api/common/RouteScope.scala
package kpn.api.common

import enumeratum.Enum
import enumeratum.EnumEntry

sealed trait RouteScope extends EnumEntry

object RouteScope extends Enum[RouteScope] {

  val all: Seq[RouteScope] = Seq(local, regional, national, international)

  val values: IndexedSeq[RouteScope] = findValues

  case object local extends RouteScope

  case object regional extends RouteScope

  case object national extends RouteScope

  case object international extends RouteScope

  case object unknown extends RouteScope
}
